### PR TITLE
Update Stable Cadence feature branch to Cadence v1.0.0-M4

### PR DESCRIPTION
**Pull request by turbolent**
______

For contributor use:

- [ ] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-cli/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Review by ianthpun — APPROVED** [review]

**Comment by turbolent** [comment]
Except for disabling a test that requires WIP code in [679b299](https://github.com/onflow/flow-cli/pull/1366/commits/679b29914a20d3885d3f6623c19b859572ad6629), this PR should be good to go now

**Review by dsainati1 — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -260,15 +258,12 @@ require (
...
+	modernc.org/memory v1.7.2 // indirect
+	modernc.org/sqlite v1.28.0 // indirect
 	nhooyr.io/websocket v1.8.7 // indirect
 	rsc.io/tmplfunc v0.0.3 // indirect
 )
 
 replace github.com/onflow/flow-cli/flowkit => ./flowkit
-
-// this is a woraround for the flow-go/crypto cross compliation issue
-replace github.com/onflow/flow-go/crypto => github.com/onflow/flow-go/crypto v0.24.9
```
Comment (tarakby) [inline go.mod:274]: A similar `replace` is still needed if you are planning to release a version from this branch.  The reason is that the CLI build process didn't get updated to manage cross compilation yet (like what I did for the [emulator](https://github.com/onflow/flow-emulator/pull/548)).

In this PR, The statement can become:
``` replace github.com/onflow/crypto => github.com/onflow/crypto v0.24.9```

If no release is needed then it's fine to remove this statement.

**Review by tarakby — COMMENTED** [review]
I only looked at the `go.mod` files and left a comment below.

File: go.mod (hunk 1 above)
Comment (turbolent) [reply go.mod:274]: Thanks for having a look! I've added the replace statement back in ab46a892db1de6b8e74bc29b8eb738e3e710b21c

**Review by bjartek — APPROVED** [review]

**Pull request merged**